Orange/widgets/visualize/owscatterplot.py

```diff
@@ -428,25 +427,26 @@ def selection_changed(self):
         self.commit()
 
     def send_data(self):
-        selected = None
-        selection = None
         # TODO: Implement selection for sql data
+        def _get_selected():
+            if not len(selection):
+                return None
+            return create_groups_table(data, graph.selection, False, "Group")
+
+        def _get_annotated():
```

Comment:
Let's consider moving this somewhere else. Code in this function is used in many widgets.